components/shopify/sources/new-paid-order/new-paid-order.js

```diff
@@ -1,23 +1,67 @@
-// Uses Shopify GraphQL API rather than REST to get Order Transactions in the same request as Orders
-// Order Transaction created_at dates are used to determine if an order's update was from being paid
-// to avoid duped orders if there are more than 100 orders between updates of a paid order
-const common = require("../common/graphql-orders.js");
+const shopify = require("../../shopify.app.js");
+const { dateToISOStringWithoutMs } = require("../common/utils");
+// Uses Shopify GraphQL API to get Order Transactions (`created_at`) in the same request as Orders (`updated_at`)
+// Order Transaction `created_at` dates are used to determine if a `PAID` Order's update was from being paid
+// to avoid duped orders if there are more than 100 orders between updates of a `PAID` Order
+// Data from GraphQL request is used to generate list of relevant Order IDs
+// Relevent Orders are requested via Shopify REST API using list of relevant Order IDs
+
+const DEFAULT_TIMER_INTERVAL_SECONDS = 60 * 5; // 5 minutes
+const MIN_ALLOWED_TRANSACT_TO_ORDER_UPDATE_MS = 1000 * 30 * 1; // 30 seconds (from testing, an Order (financial_status:=PAID) seems to be updated within 15s of the Transaction)
 
 module.exports = {
-  ...common,
   key: "shopify-new-paid-order",
   name: "New Paid Order",
   description: "Emits an event each time a new order is paid.",
-  version: "0.0.1",
+  version: "0.0.2",
   dedupe: "unique",
   props: {
-    ...common.props,
+    db: "$.service.db",
+    timer: {
+      type: "$.interface.timer",
+      default: {
+        intervalSeconds: DEFAULT_TIMER_INTERVAL_SECONDS,
+      },
+    },
+    shopify,
   },
   methods: {
-    ...common.methods,
+    _getPrevCursor() {
+      return this.db.get("prevCursor") || null;
+    },
+    _setPrevCursor(prevCursor) {
+      this.db.set("prevCursor", prevCursor);
+    },
+    /**
+     * Returns the allowed time interval between Order Transaction and Order
+     * update where the Order update can still be considered the result of
+     * becoming 'paid'.
+     *
+     * @returns {number} The number of milliseconds after a Transaction is created
+     */
+    _getAllowedTransactToOrderUpdateMs() {
+      return Math.max(
+        MIN_ALLOWED_TRANSACT_TO_ORDER_UPDATE_MS,
+        2 * DEFAULT_TIMER_INTERVAL_SECONDS * 1000
+      );
+    },
+    getQueryFields() {
+      // See https://shopify.dev/docs/admin-api/graphql/reference/orders/order
+      // With these fields, queries cost up to 202 points
+      return [
+        "updatedAt",
+        "name",
+        "legacyResourceId", // The ID of the corresponding resource (Order) in the REST Admin API.
+        "transactions.createdAt",
+      ];
+    },
+    getQueryFilter({ updatedAt = null }) {
+      return `financial_status:paid updated_at:>${updatedAt}`;
+    },
```

Comment:
Can you document these 2 methods?
It's likely that they'll be moved to a parent/common component as more and more GraphQL-based components are developed.

Thanks!